src/CustomIndexing/CustomIndexing.jl

```diff
@@ -0,0 +1,130 @@
+module CustomIndexing
+
+"""
+    struct SparseArray{T,N} <: AbstractArray{T,N}
+        data::Dict{Tuple,T}
+    end
+
+`N`-dimensional array with elements of type `T` where only a subset of the entries are defined.
+The entries with indices `idx = (i1, i2, ..., iN)` in `keys(data)` has value `data[idx]`.
+Note that as opposed to `SparseArrays.AbstractSparseArray`, the missing entries are not assumed
+to be `zero(T)`. The indexing structure of the array is not rectangular and the missing entries
```

Comment:
"not rectangular" is a bit unclear. I think "The missing entries are simply not part of the array." says the same thing.